geozero/src/geojson/geojson_writer.rs

```diff
@@ -192,7 +192,10 @@ fn write_num_prop<'a, W: Write>(out: &'a mut W, colname: &str, v: &dyn Display)
 }
 
 fn write_str_prop<'a, W: Write>(out: &'a mut W, colname: &str, v: &dyn Display) -> Result<()> {
-    out.write_all(&format!(r#""{}": "{}""#, colname, v).as_bytes())?;
+    let esc_colname: String = colname.replace("\"", "\\\"");
+    let esc_value: String = format!("{}", v).replace("\"", "\\\"");
```

Comment:
I'm not sure, whether this additional String allocation is optimzied away by the compiler.
To be sure, I would escape directly in the `write_all` call.

Reply:
I wasn't sure how to escape `v` without using `format!()` because `v` is `Display` trait, not `str`. How would you do this without allocating a new string?

Reply:
So
`out.write_all(&format!(r#""{}": "{}""#, colname.replace("\"", "\\\""), v.replace("\"", "\\\"")).as_bytes())?;`
doesn't compile?

Reply:
Right, I get this:

```rust
error[E0599]: no method named `replace` found for reference `&dyn std::fmt::Display` in the current scope
   --> geozero/src/geojson/geojson_writer.rs:201:11
    |
201 |         v.replace("\"", "\\\"")
    |           ^^^^^^^ method not found in `&dyn std::fmt::Display`
```

Happy to check out other options if you know of any.

Reply:
This compiles:

```diff
 fn write_num_prop<'a, W: Write>(out: &'a mut W, colname: &str, v: &dyn Display) -> Result<()> {
-    out.write_all(&format!(r#""{}": {}"#, colname, v).as_bytes())?;
+    out.write_all(&format!(r#""{}": {}"#, colname.replace("\"", "\\\""), v).as_bytes())?;
     Ok(())
 }
 
-fn write_str_prop<'a, W: Write>(out: &'a mut W, colname: &str, v: &dyn Display) -> Result<()> {
-    out.write_all(&format!(r#""{}": "{}""#, colname, v).as_bytes())?;
+fn write_str_prop<'a, W: Write>(out: &'a mut W, colname: &str, v: &str) -> Result<()> {
+    out.write_all(
+        &format!(
+            r#""{}": "{}""#,
+            colname.replace("\"", "\\\""),
+            v.replace("\"", "\\\"")
+        )
+        .as_bytes(),
+    )?;
     Ok(())
 }
```

Reply:
Just pushed your suggested changes, thanks for figuring that out